test/test_ecco4_en4.jl

```diff
@@ -194,45 +172,45 @@ end
-        @test @allowscalar T_restoring(1, 90,  10, grid, clock, fields) == T_restoring.rate / 2
-        @test @allowscalar T_restoring(1, 100, 10, grid, clock, fields) == T_restoring.rate
-        @test @allowscalar T_restoring(1, 1,   5,  grid, clock, fields) == 0
-        @test @allowscalar T_restoring(1, 10,  5,  grid, clock, fields) == 0
+        for name in (:temperature, :salinity)
+            var_restoring = DatasetRestoring(name; dataset, start_date, end_date, mask, inpainting, rate=1/1000)
+
+            fill!(var_restoring.field_time_series[1], 1.0)
+            fill!(var_restoring.field_time_series[2], 1.0)
+
+            T = CenterField(grid)
+            S = CenterField(grid)
+            fields = (; T, S)
+            clock  = Clock(; time = 0)
+
+            @test @allowscalar var_restoring(1, 1,   10, grid, clock, fields) == var_restoring.rate
+            @test @allowscalar var_restoring(1, 11,  10, grid, clock, fields) == var_restoring.rate / 2
+            @test @allowscalar var_restoring(1, 21,  10, grid, clock, fields) == 0
+            @test @allowscalar var_restoring(1, 80,  10, grid, clock, fields) == 0
+            @test @allowscalar var_restoring(1, 90,  10, grid, clock, fields) == var_restoring.rate / 2
+            @test @allowscalar var_restoring(1, 100, 10, grid, clock, fields) == var_restoring.rate
+            @test @allowscalar var_restoring(1, 1,   5,  grid, clock, fields) == 0
+            @test @allowscalar var_restoring(1, 10,  5,  grid, clock, fields) == 0
+        end
     end
-end
-
-@testset "Timestepping with ECCORestoring" begin
-    for arch in test_architectures
 
-        grid  = LatitudeLongitudeGrid(arch;
-                                      size = (10, 10, 10),
-                                      latitude = (-60, -40),
-                                      longitude = (10, 15),
-                                      z = (-200, 0),
-                                      halo = (6, 6, 6))
+    @testset "Timestepping with DatasetRestoring" begin
+        grid = LatitudeLongitudeGrid(arch;
+                                     size = (10, 10, 10),
+                                     latitude = (-60, -40),
+                                     longitude = (10, 15),
+                                     z = (-200, 0),
+                                     halo = (6, 6, 6))
 
         field = CenterField(grid)
 
         @test begin
-            set!(field, ECCOMetadatum(:temperature, date=start_date))
-            set!(field, ECCOMetadatum(:salinity,    date=start_date))
+            set!(field, Metadatum(:temperature; dataset, date=start_date))
+            set!(field, Metadatum(:salinity;    dataset, date=start_date))
             true
         end
 
-        forcing_T = ECCORestoring(:temperature, arch; start_date, end_date, inpainting, rate=1/1000)
+        forcing_T = DatasetRestoring(:temperature;  dataset, start_date, end_date, inpainting, rate=1/1000)
```

Comment:
```suggestion
        forcing_T = DatasetRestoring(:temperature, arch; dataset, start_date, end_date, inpainting, rate=1/1000)
```

Reply:
oh I see! we dropped `arch`

wondering how come tests pass on tartarus GPU... I'll fix this

Reply:
Thanks!